src/services/measure/measure.js

```diff
@@ -0,0 +1,270 @@
+   * @private
+   */
+  this.helpTooltip_;
+
+
+  /**
+   * The measure tooltip element.
+   * @type {Element}
+   * @private
+   */
+  this.measureTooltipElement_;
+
+
+  /**
+   * Overlay to show the measurement.
+   * @type {ol.Overlay}
+   * @private
+   */
+  this.measureTooltip_;
+
+
+  /**
+   * Pointer move listener key
+   * @type {goog.events.Key}
+   * @private
+   */
+  this.pointerMoveHandlerKey_;
+
+
+  /**
+   * The sketch feature.
+   * @type {ol.Feature}
+   */
+  this.sketchFeature;
+
+  /**
+   * The draw interaction to be used.
+   * @type {ol.interaction.Draw|ngeo.interaction.DrawAzimut}
+   */
+  this.drawInteraction = null;
+
+  goog.base(this);
+
+
+  /**
+   * The draw overlay
+   * @type {ol.FeatureOverlay}
+   */
+  this.overlay = new ol.FeatureOverlay({
+    style: [
+      new ol.style.Style({
+        fill: new ol.style.Fill({
+          color: 'rgba(255, 255, 255, 0.2)'
+        })
+      }),
+      new ol.style.Style({
+        stroke: new ol.style.Stroke({
+          color: 'white',
+          width: 5
+        })
+      }),
+      new ol.style.Style({
+        stroke: new ol.style.Stroke({
+          color: '#ffcc33',
+          width: 3
+        })
+      })
+    ]
+  });
+};
+goog.inherits(ngeo.interaction.Measure, ol.interaction.Pointer);
+
+
+/**
+ * @inheritDoc
+ */
+ngeo.interaction.Measure.prototype.setMap = function(map) {
+  goog.base(this, 'setMap', map);
+  this.overlay.setMap(map);
+
+  map.addInteraction(this.drawInteraction);
+
+  this.drawInteraction.on('drawstart', goog.bind(this.onDrawStart_, this));
+  this.drawInteraction.on('drawend', goog.bind(this.onDrawEnd_, this));
+};
+
+
+/**
+ * Handle draw interaction `drawstart` event.
+ * @param {ol.DrawEvent} evt
+ * @private
+ */
+ngeo.interaction.Measure.prototype.onDrawStart_ = function(evt) {
+  this.sketchFeature = evt.feature;
+  this.overlay.getFeatures().clear();
+  this.createMeasureTooltip_();
+};
+
+
+/**
+ * Handle draw interaction `drawend` event.
+ * @param {ol.DrawEvent} evt
+ * @private
+ */
+ngeo.interaction.Measure.prototype.onDrawEnd_ = function(evt) {
+  // FIXME we shouldn't manipulate DOM style here
+  this.measureTooltipElement_.className = 'tooltip tooltip-static';
+  this.measureTooltip_.setOffset([0, -7]);
+  this.sketchFeature = null;
+};
+
+
+/**
+ * Creates a new help tooltip
+ * @private
+ */
+ngeo.interaction.Measure.prototype.createHelpTooltip_ = function() {
+  this.removeHelpTooltip_();
+  this.helpTooltipElement_ = document.createElement('div');
+  this.helpTooltipElement_.className = 'tooltip';
+  this.helpTooltip_ = new ol.Overlay({
+    element: this.helpTooltipElement_,
+    offset: [15, 0],
+    positioning: 'center-left'
+  });
+  this.getMap().addOverlay(this.helpTooltip_);
+};
+
+
+/**
+ * Destroy the help tooltip
+ * @private
+ */
+ngeo.interaction.Measure.prototype.removeHelpTooltip_ = function() {
+  this.getMap().removeOverlay(this.helpTooltip_);
+  if (this.helpTooltipElement_) {
```

Comment:
And the same at some other places. Usually we explicitly specify what we are checking for, e.g. `isDef` or `isNull`.